### PR TITLE
contrib: add raw release notes

### PR DESCRIPTION
**Pull request by mattklein123**

### CODE REVIEW

**Comment by mattklein123** [comment]
@envoyproxy/maintainers please review. This is the last release I'm manually doing release notes for. :)

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -33,16 +33,16 @@ components, etc.
...
 if you are unsure. A good rule of thumb is the riskier the change, the
 more comprehensive the testing should be.
 
 [optional *Deprecated*:]
 > Description of what is deprecated.
 
-[optional *Release Notes*:]
- > Description of what to add to next point release notes.
+[optional *Release Notes*:] If this change is user impacting you **must** add a release note to
+[RAW_RELEASE_NOTES.md](RAW_RELEASE_NOTES.md). Thank you!
```
Comment (alyssawilk) [inline PULL_REQUEST_TEMPLATE.md]: Would it be a bunch of work to have a presubmit which required you either edit RAW_RELEASE_NOTES or have metadata in your PR explicitly noting that you don't need to?  Might be worth a filed issue if we find ourselves forgetting.  presubmits like this are annoying as heck but do tend to catch a lot.

Comment (htuch) [reply PULL_REQUEST_TEMPLATE.md]: There already is an optional metadata field in the commit for this, it's a question of enforcing.

File: RAW_RELEASE_NOTES.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Raw release notes
+
+This file contains "raw" release notes for each release. The notes are added by developers as changes
+are made to Envoy that are user impacting. When a release is cut, the releaser will use these notes
+to populate the [Sphinx release notes in data-plane-api](https://github.com/envoyproxy/data-plane-api/blob/master/docs/root/intro/version_history.rst).
+The idea is that this is a low friction way to add release notes along with code changes. Doing this
+will make it substantially easier for the releaser to "linkify" all of the release notes in the
+final version.
```
Comment (htuch) [inline RAW_RELEASE_NOTES.md]: I think we can populate this from the git commit logs if we are diligent about ensuring that folks include the release notes field (and make sure it's machine parseable).

File: PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (mattklein123) [reply PULL_REQUEST_TEMPLATE.md]: That sounds awesome, but I would rather walk before we run. I will open a follow up issue.

Comment (mattklein123) [reply PULL_REQUEST_TEMPLATE.md]: Opened https://github.com/envoyproxy/envoy/issues/2133

**Pull request merged**